clients/ts/signalr/src/LongPollingTransport.ts

```diff
@@ -177,29 +162,39 @@ export class LongPollingTransport implements ITransport {
     }
 
     public async stop(): Promise<void> {
-        // Send a DELETE request to stop the poll
+        this.logger.log(LogLevel.Trace, "(LongPolling transport) Stopping polling.");
+
+        // Tell receiving loop to stop, abort any current request, and then wait for it to finish
+        this.running = false;
+        this.pollAbort.abort();
+
         try {
-            this.running = false;
+            await this.receiving;
+
+            // Send DELETE to clean up long polling on the server
             this.logger.log(LogLevel.Trace, `(LongPolling transport) sending DELETE request to ${this.url}.`);
 
             const deleteOptions: HttpRequest = {
                 headers: {},
             };
             const token = await this.accessTokenFactory();
             this.updateHeaderToken(deleteOptions, token);
-            const response = await this.httpClient.delete(this.url, deleteOptions);
+            await this.httpClient.delete(this.url, deleteOptions);
 
-            this.logger.log(LogLevel.Trace, "(LongPolling transport) DELETE request accepted.");
+            this.logger.log(LogLevel.Trace, "(LongPolling transport) DELETE request sent.");
         } finally {
-            // Abort the poll after the shutdown timeout if the server doesn't stop the poll.
-            if (!this.stopped) {
-                this.shutdownTimer = setTimeout(() => {
-                    this.logger.log(LogLevel.Warning, "(LongPolling transport) server did not terminate after DELETE request, canceling poll.");
-
-                    // Abort any outstanding poll
-                    this.pollAbort.abort();
-                }, this.shutdownTimeout);
-            }
+            this.logger.log(LogLevel.Trace, "(LongPolling transport) Stop finished.");
+
+            // Raise close event here instead of in polling
+            // It needs to happen after the DELETE request is sent
+            this.raiseOnClose();
+        }
+    }
+
+    private raiseOnClose() {
+        if (this.onclose) {
+            this.logger.log(LogLevel.Trace, `(LongPolling transport) Firing onclose event. Error: ${this.closeError || "<undefined>"}`);
```

Comment:
Graceful closes will log "Error: \<undefined\>"?

Reply:
Yeah. That's what it currently logs.

Reply:
Could we change it to not have "Error: \<undefined\>" for clean close